### PR TITLE
WT-7195 Remove non-ascii input and paired typos in any source files

### PR DESCRIPTION
**Pull request by fionaochee**
* Add non-ascii character removal on `.dox`, `.cxx`, `.py`, `.cmake`,  `.i` source files in `s_style`.
* Remove similar check in `s_docs` which prints error at non-ascii characters in `.dox` files.
* Add check for paired typos to `.cxx`, `.py`, `.cmake`,  `.i` source files.
* Fix paired typos in source files.

### CODE REVIEW

File: dist/s_style [hunk 1]
```diff
@@ -17,8 +17,9 @@ if [ $# -ne 1 ]; then
 	cd ..
 
 	find bench examples ext src test \
-	    -name '*.[chsy]' -o -name '*.in' -o -name '*.dox' |
+	    -name '*.[chsy]' -o -name '*.in' -o -name '*.dox' -o -name '*.cxx' -o -name '*.py' -o -name '*.cmake' -o -name '*.i' |
 	sed -e '/Makefile.in/d' \
+		-e '/test\/3rdparty/d' \
```
Comment (etienneptl) [inline dist/s_style:22]: np: Syntax issue

File: dist/s_docs [hunk 2]
```diff
@@ -90,20 +90,6 @@ structurechk()
 	}
 }
 
-asciichk()
-{
-    # Make sure there are only ASCII characters in the input files.
-    # Odd characters can cause our Python filter to fail.
-    (cd ../src/docs && LC_ALL=C grep -n '[^[:print:]	]' *.dox | cat -v) > $t
```
Comment (etienneptl) [inline dist/s_docs:97]: Do we no longer need this? Is it covered by what's in `s_style`?

File: dist/s_style [hunk 3]
```diff
@@ -48,7 +49,15 @@ else
 		fi
 	fi
 
-	egrep -w 'a a|an an|and and|are are|be be|by by|for for|from from|if if|in in[^-]|is is|it it|of of|the the|this this|to to|was was|were were|when when|with with|a an|an a|a the|the a' $f > $t
+	tr -cd '[:alnum:][:space:][:punct:]' < $f |
```
Comment (etienneptl) [inline dist/s_style:52]: Can we add a comment explaining what this section of code does?

**Review by etienneptl — COMMENTED** [review]
Good job on catching (and fixing !) all the mistakes left behind. I left a couple of comments and a question.

File: dist/s_docs (hunk 2 above)
Comment (fionaochee) [reply dist/s_docs:97]: Yes, this work overlaps with what's covered in `s_style` now, as this outputs an error for non-ascii character in `.dox` files, and s_style will automatically remove the non-ascii characters in `.dox` files now.

**Review by ddanderson — APPROVED** [review]
LGTM

File: dist/s_docs (hunk 2 above)
Comment (etienneptl) [reply dist/s_docs:97]: Thanks for the details @fionaochee !

**Review by etienneptl — APPROVED** [review]
LGTM, thanks for looking into this!

**Review by jiechenbo — APPROVED** [review]
LGTM! Thanks for doing this!

**Pull request merged**